### PR TITLE
updated cutdown driver with cutdown format

### PR DESCRIPTION
**Pull request by IonLi09**
added cutdown sequence number to logger.

### CODE REVIEW

File: EosPayload/drivers/cutdown_driver.py [hunk 1]
```diff
@@ -76,6 +78,12 @@ def cutdown_trigger(self):
 
     @staticmethod
     def cutdown_trigger_mqtt(client, user_data, message):
-        user_data['logger'].info("received cutdown command")
-        user_data['queue'].put(1)
+        try:
+            decoded_msg = CutDown.decode(message)
+            if decoded_msg.get_format_type() != Type.CutDown:
+                raise TypeError("incorrect type")
```
Comment (DarylDohner) [inline EosPayload/drivers/cutdown_driver.py:84]: don't throw, you should instead use logger.error()
also consider a more descriptive message like "incorrect type X, expected CutDown"

File: EosPayload/drivers/cutdown_driver.py [hunk 2]
```diff
@@ -76,6 +78,12 @@ def cutdown_trigger(self):
 
     @staticmethod
     def cutdown_trigger_mqtt(client, user_data, message):
-        user_data['logger'].info("received cutdown command")
-        user_data['queue'].put(1)
+        try:
+            decoded_msg = CutDown.decode(message)
```
Comment (DarylDohner) [inline EosPayload/drivers/cutdown_driver.py:82]: so clearly this wasnt tested...you need Packet.decode here

File: EosPayload/drivers/cutdown_driver.py [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 from EosPayload.lib.base_drivers.position_aware_driver_base import PositionAwareDriverBase
 from EosPayload.lib.mqtt import Topic
+from EosLib.format.formats.cutdown import CutDown
```
Comment (DarylDohner) [inline EosPayload/drivers/cutdown_driver.py:7]: you need to update requirements.txt to the correct EosLib version

**Review by DarylDohner — DISMISSED** [review]
a few things
1. see comments
2. you need to actually test this, i would recommend either writing a quick script that sends a cutdown command over mqtt or just hooking up the radio / ground station, not sure how far along that is
3. there are a few places in the cutdown driver that call time.sleep, please replace these with self.thread_sleep

File: EosPayload/drivers/cutdown_driver.py [hunk 4]
```diff
@@ -60,22 +66,44 @@ def device_read(self, logger: logging.Logger) -> None:
...
 
-    @staticmethod
-    def cutdown_trigger_mqtt(client, user_data, message):
-        user_data['logger'].info("received cutdown command")
-        user_data['queue'].put(1)
+    def cutdown_trigger_mqtt(self, client, user_data, message):
+        try:
+            packet = Packet.decode(message.payload)
+            if packet.data_header.data_type != Type.CUTDOWN:
+                user_data['logger'].error(f"incorrect type {packet.data_header.data_type}, expected CutDown")
```
Comment (VanL15) [inline EosPayload/drivers/cutdown_driver.py:87]: Make the 'i' in 'incorrect' capital

File: EosPayload/drivers/cutdown_driver.py [hunk 5]
```diff
@@ -60,22 +66,44 @@ def device_read(self, logger: logging.Logger) -> None:
...
+                data_type=Type.CUTDOWN,
+                sender=self.get_device_id(),
+                priority=Priority.URGENT,
+                destination=packet.data_header.sender
+            )
+
+            response = Packet(CutDown(decoded_msg.ack), response_header)
+            client.send(Topic.RADIO_TRANSMIT, response)
+
+            user_data['logger'].info(f"received ACK for cutdown from device '{packet.data_header.sender}'"
```
Comment (VanL15) [inline EosPayload/drivers/cutdown_driver.py:105]: make the 'r' in 'received' capital

File: EosPayload/drivers/radio_driver.py [hunk 6]
```diff
@@ -89,12 +90,13 @@ def data_receive_callback(xbee_message):
...
             logger.info("Packet received ~~~~~~")
             logger.info(packet)
-            packet_object = Packet.decode(packet)  # convert packet bytearray to packet object
+            packet_object = Packet.decode(bytes(packet))  # convert packet bytearray to packet object
 
             # Try to data log the packet, but we really don't want to block in a callback
             if self.log_lock.acquire(blocking=False):
                 try:
-                    self.data_log(["received", packet_object.encode_to_string()])
+                    #self.data_log(["LMAO THIS DOESN'T WORK but here is dest: "])#, packet_object.data_header.destination])
```
Comment (VanL15) [inline EosPayload/drivers/radio_driver.py:98]: Remove this comment

**Review by VanL15 — CHANGES_REQUESTED** [review]
see comments, wait until ping is merged

**Review by aryanbattula — DISMISSED** [review]
good job!

**Review by aryanbattula — APPROVED** [review]
lgtm

**Pull request merged**